wlutil/br/br.py

```diff
@@ -37,6 +38,55 @@
 
 exit""")
 
+def buildConfig():
+    """Construct the final buildroot configuration for this environment. After
+    calling this, it is safe to call 'make' in the buildroot directory."""
+
+    # We run the preprocessor on a simple program to see the C-visible
+    # "LINUX_VERSION_CODE" macro
+    linuxHeaderTest = """#include <linux/version.h>
+    LINUX_VERSION_CODE
+    """
+    linuxHeaderVer = sp.run(['riscv64-unknown-linux-gnu-gcc', '-E', '-xc', '-'],
+              input=linuxHeaderTest, stdout=sp.PIPE, universal_newlines=True)
+    linuxHeaderVer = linuxHeaderVer.stdout.splitlines()[-1].strip()
+
+    # Major/minor version of the linux kernel headers included with our
+    # toolchain. This is not necessarily the same as the linux kernel used by
+    # Marshal, but is assumed to be <= to the version actually used.
+    linuxMaj = str(int(linuxHeaderVer) >> 16)
+    linuxMin = str((int(linuxHeaderVer) >> 8) & 0xFF)
+
+    # Toolchain major version
+    toolVerStr = sp.run("riscv64-unknown-linux-gnu-gcc --version", shell=True,
```

Comment:
This also doesn't need to be a shell command.  The added benefit is that `subprocess.run` raises `FileNotFoundError` (ENOENT) if the executable isn't in `PATH`, rather than having to explicitly check the return code.

Reply:
done